### ISSUE TITLE
Enabling settings to override by language

### ISSUE THREAD

**Issue by ufukty** [lorefnon/lowlight-patterns#1]
Problem:

```json
{
    "lowlight-patterns.opacity": 0.33,
    "[json]": {
        "lowlight-patterns.rules": [".*\\$schema.*"],
    }
    "[go]": {
        "lowlight-patterns.rules": [".*\\!= err.*"],
    }
}
```

Currently when language-specific setting are applied, it raises an error such:

`This setting cannot be applied because it is not registered as language override setting.`

It could be much better to define language specific regexes. Since, they are not context-aware and there could be collisions of wanted and unwanted text between different languages.

**Comment by lorefnon** [comment]
Yup, we'd definitely want to have this. Would you be interested in sending a pr to support this ? 

If not I'd take a look at this sometime next week or so.

**Comment by ufukty** [comment]
I will take a look

**Comment by ufukty** [comment]
It looks like there is a value called `language-overridable` for `scope` prop of extension parameters. 

I could not figure out how to access language overriden config. `vscode.workspace.getConfig("lowlight-patterns", editor.document.uri)`still gives the variables defined in workspace and ignores language specific rules.
